waiter/src/waiter/scheduler.clj

```diff
@@ -642,3 +647,192 @@
+            ;; React to upward-scaling
+            instance-counts' (get service-id->instance-counts service-id instance-counts-zero)
+            instances-requested-delta (- (:requested instance-counts')
+                                         (:requested instance-counts))
+            instance-scheduling-start-times'
+            (->> current-time
+                 (repeat instances-requested-delta)
+                 (into instance-scheduling-start-times))
+            ;; Track scheduling instances
+            unmatched-instance-count (- (count new-instance-ids)
+                                        (count instance-scheduling-start-times'))
+            [matched-start-times instance-scheduling-start-times'']
+            (split-at (count new-instance-ids) instance-scheduling-start-times')
+            ;; Track starting instances
+            new-instance-id->start-timestamp
+            (pc/map-from-keys (constantly current-time) new-instance-ids)
+            starting-instance-id->start-timestamp'
+            (->> removed-instance-ids
+                 (reduce dissoc starting-instance-id->start-timestamp)
+                 (merge new-instance-id->start-timestamp))
+            {started-instance-ids :started starting-instances-ids :starting}
+            (group-by #(if (healthy-instance-id? %) :started :starting)
+                      (keys starting-instance-id->start-timestamp'))
+            starting-instance-id->start-timestamp''
+            (select-keys starting-instance-id->start-timestamp' starting-instances-ids)]
+        ;; Warn about any new instances for which we didn't track the scheduling time
+        (when (pos? unmatched-instance-count)
+          (log/warn unmatched-instance-count
+                    "untracked scheduled instances discovered in instance launch metric loop for service"
+                    service-id))
+        ;; Report schedule time for now-known instances
+        (doseq [start-time matched-start-times]
+          (let [duration (metrics/duration-between start-time current-time)]
+            (metrics/report-duration waiter-schedule-timer duration)
+            (metrics/report-duration service-schedule-timer duration)
+            (statsd/timing! metric-group "schedule_time" duration)))
+        ;; Report startup time for instances that are now healthy
+        (doseq [instance-id started-instance-ids]
+          (let [start-time (starting-instance-id->start-timestamp' instance-id)]
+            (metrics/report-duration service-startup-timer start-time current-time)))
+        ;; tracker-state'
+        (assoc tracker-state
+               :instance-counts instance-counts'
+               :instance-scheduling-start-times (vec instance-scheduling-start-times'')
+               :known-instance-ids known-instance-ids'
+               :starting-instance-id->start-timestamp starting-instance-id->start-timestamp'')))))
+
+(defmacro build-inital-service-launch-trackers
```

Comment:
Nitpick: typo in `inital`

Reply:
Missing docstring

Reply:
Added docstring. It looks like I fixed the `initial` typo in an earlier commit.

Reply:
The typo is still there :(

Reply:
You're right, there was still a typo. Fixed.